File: backend/equis/src/main/java/es/urjc/daw/equis/dto/PostDTO.java
package es.urjc.daw.equis.dto;

public record PostDTO(
    Long id,
    String content,
    String createdAt,
    long likesCount,
    int commentsCount,

    // user (basic info)
    Long userId,
    String userNickname,

    // category (basic info)
    Long categoryId,
    String categoryName
) {}

File: frontend/app/dto/PostDTO.ts
export type CommentDTO = {
  id: number;
  content: string;
  createdAt: string;
  likesCount: number;
  userId: number;
  userNickname: string;
  likedByCurrentUser: boolean;
};

export type PostDTO = {
  id: number;
  content: string;
  createdAt: string;
  likesCount: number;
  userId: number;
  userNickname: string;
  categoryId: number;
  categoryName: string;
  comments: CommentDTO[];
  likedByCurrentUser: boolean;
};

export type CategoryDTO = {
  id: number;
  name: string;
  description?: string;
  imageUrl?: string;
  postsCount?: number;
};
